Simplicity.C.nix

```diff
@@ -1,13 +1,22 @@
-{ lib, stdenv, wideMultiply ? null }:
+{ lib, stdenv, gcovr ? null, wideMultiply ? null, withCoverage ? false
+, gcov-executable ? "gcov" # Set to "${nixpkgs.libllvm}/bin/llvm-cov gcov" when using clang.
```

Comment:
I'd suggest wrapping `gcovr` and `gcov-executable` into a set (though have the whole thing default to `null`) and then passing either a set or `null` in.

Then in default.nix the user can pass either a set or `null`, and it will be forwarded.

I've been able to run this just by passing `withCoverage` as `true` into default.nix, and confirmed that it produces some sort of coverage output (though I didn't look at it), so I"m kinda confused about what these extra variables are...

Reply:
I've reworked this line.